Shared/toast.php

```diff
@@ -102,6 +102,54 @@ function toast(type, text, duration, arguments = null) {
         // Makes the toast visible ("show" is a class that makes the toast visible through css)
         toastDiv.classList.add('show', 'toast');
 
+        function truncateOverflow(toastText) {
+        const maxChars = 70; // Set the maximum number of characters before truncating
+        const toastContent = toastText.textContent.trim();
+        if (toastContent.length > maxChars) {
+            // Truncate the content and add "..."
+            toastText.textContent = toastContent.slice(0, maxChars) + '...';
+            toastText.dataset.fullText = toastContent; // Store the full text in a data attribute
+            }
+        }
+
+
+
+
+        toastDiv.addEventListener('click', function(event) {
+            // Check if the click target is not the close button
+            if (!event.target.classList.contains('closeIcon')) {
+
+
+                const toastText = this.querySelector('.toastText');
+                const hasOverflow = toastText.scrollHeight > toastText.clientHeight;
+                const isExpanded = this.classList.contains('expanded');
+
+
+                this.classList.toggle('expanded');
+                toastText.textContent = toastText.dataset.fullText || toastText.textContent;
+                   
+                    // If the toast is expanded, set the height to auto to show full text
+                    if (this.classList.contains('expanded'))
+                    {
+                        // Reset the height to auto to show full text
+                        toastText.style.height = 'auto';
+                    }
+                      // If the toast was previously expanded and is now collapsed, reset the height to 2rem
+                    if (isExpanded && !this.classList.contains('expanded'))
+                    {
+                        truncateOverflow(toastText); // Truncate overflow text
+                        toastText.style.height = '2.3rem';
+                    }
+
+
+            }
+        });
+
+
+        truncateOverflow(toastText); // Truncate overflow text initially
+
+
+
```

Comment:
I won't comment for every unnecessary white space, but you should generally get rid of all of them